src/main/java/com/puppycrawl/tools/checkstyle/site/ClassAndSettersJavadocScraper.java

```diff
@@ -0,0 +1,155 @@
+
+    /** The module javadoc. */
+    private static DetailNode moduleJavadoc;
+
+    /** Map of the module property setter javadocs. */
+    private static final Map<String, DetailNode> MODULE_PROPERTY_SETTER_JAVADOCS = new TreeMap<>();
+
+    /**
+     * Set the module name to be scraped.
+     *
+     * @param moduleName the module name.
+     */
+    public static void setModuleName(String moduleName) {
+        ClassAndSettersJavadocScraper.moduleName = moduleName;
+    }
+
+    /**
+     * Set the module javadoc.
+     *
+     * @param moduleJavadoc the module javadoc.
+     */
+    public static void setModuleJavadoc(DetailNode moduleJavadoc) {
+        ClassAndSettersJavadocScraper.moduleJavadoc = moduleJavadoc;
+    }
+
+    /** Clear the module property setter javadocs of any previous information. */
+    public static void clearModulePropertySetterJavadocs() {
+        MODULE_PROPERTY_SETTER_JAVADOCS.clear();
+    }
+
+    /**
+     * Get the module property setter javadocs.
+     *
+     * @return the module property setter javadocs map.
+     */
+    public static Map<String, DetailNode> getModulePropertySetterJavadocs() {
+        return Collections.unmodifiableMap(MODULE_PROPERTY_SETTER_JAVADOCS);
+    }
+
+    /**
+     * Get the module javadoc.
+     *
+     * @return the module javadoc.
+     */
+    public static DetailNode getModuleJavadoc() {
+        return moduleJavadoc;
+    }
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.JAVADOC,
+        };
+    }
+
+    @Override
+    public void visitJavadocToken(DetailNode ast) {
+        final DetailAST blockCommentAst = getBlockCommentAst();
+        if (BlockCommentPosition.isOnMethod(blockCommentAst)) {
+            final DetailAST methodDef = getParentAst(blockCommentAst, TokenTypes.METHOD_DEF);
+            final String methodName = methodDef.findFirstToken(TokenTypes.IDENT).getText();
+            if (CheckUtil.isSetterMethod(methodDef) && isMethodOfScrapedModule(methodDef)) {
+                MODULE_PROPERTY_SETTER_JAVADOCS.put(methodName, ast);
+            }
+        }
+        else if (BlockCommentPosition.isOnClass(blockCommentAst)) {
+            final DetailAST classDef = getParentAst(blockCommentAst, TokenTypes.CLASS_DEF);
+            final String className = classDef.findFirstToken(TokenTypes.IDENT).getText();
+            if (className.equals(moduleName)) {
+                moduleJavadoc = ast;
+            }
+        }
+    }
+
+    /**
+     * Checks if the given method is a method of the module being scraped. Traverses
+     * parent nodes until it finds the class definition and checks if the class name
+     * is the same as the module name.
+     *
+     * @param methodDef the method definition.
+     * @return true if the method is a method of the given module, false otherwise.
+     */
+    private static boolean isMethodOfScrapedModule(DetailAST methodDef) {
```

Comment:
Help me to understand what the conditions are where we can have a method that is NOT a method of the module being scraped.

Reply:
Inner class.

I updated Javadoc to make it more obvious what we are avoiding here.